### ISSUE TITLE
failed to push some refs to <repo_url>

### ISSUE THREAD

**Issue by ribtoks** [Andrew-Chen-Wang/github-wiki-action#7]
Hi

Thank you for developing this awesome GitHub Action!

My recent run has failed with this error:

```
To https://github.com/ribtoks/<my-repo>.wiki.git
 ! [rejected]        master -> master (fetch first)
error: failed to push some refs to 'https://github.com/ribtoks/xpiks.wiki.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
```

Do you have any idea how to resolve it? Might it be a good idea to pull inside your action before pushing?

Thanks,
Taras

**Comment by Andrew-Chen-Wang** [comment]
Hm do you mind retrying? Like literal retry the push? Because we already perform a pull: https://github.com/Andrew-Chen-Wang/github-wiki-action/blob/8dc1f34a2aff416ac409c278ac8a04e4a8faa0ff/entrypoint.sh#L40

**Comment by ribtoks** [comment]
I'm not sure I understood what you're suggesting. I pushed a commit that had changes in source code files and documentation (in markdowns, "wiki material"). My commit is already in the repository. What do you mean by "retry the push"?

If you mean "retry the action run itself", I did that and it failed repeatedly.

**Comment by Andrew-Chen-Wang** [comment]
tl;dr Unfortunately, I don't really know how to reproduce this. Fortunately, this GitHub action is really simple that you can easily replicate on your local machine.

Sorry for the wait and miscommunication. Yes, I meant retry the action. A possibility for this bug is that GitHub changed the default branch for wikis. I just tested it, and GitHub still uses default master branch, though you may still want to check. If your branch is not master, then I can add an input to change the wiki branch we push to.

We may need to just use the flag `--allow-unrelated-histories` since we already save wiki history in the main repo anyways. But I'd like to avoid that.
